sycl/include/CL/sycl/handler.hpp

```diff
@@ -820,6 +899,12 @@ class __SYCL_EXPORT handler {
 
 #endif
 
+  /// Check if the queue being used is for a GPU device
+  ///
+  /// \param Queue is the queue for this handler.
+  /// \return Whether the device is a GPU.
+  bool is_gpu(shared_ptr_class<sycl::detail::queue_impl> Queue);
```

Comment:
Can we rename this to something like `requires_uniform_groups` or `prefers_rounding`?  Motivation:

1) It's not clear to me that this optimization really is only useful on GPUs.  In my testing, I saw an improvement on CPUs as well (although it was smaller).

2) It's not clear from the name `is_gpu` that this function is related to the functionality being added here.  It'll be easier for people to find this function and enable this optimization for other classes of devices in the future if the names are linked.

3) If we're concerned about potential ABI changes, adopting a more general name now seems more future-proof.  If I'm right that other devices may want to use this optimization, moving from `is_gpu` to something else later will require a second round of ABI changes.

Reply:
1. This optimization addresses the situation where the GPU driver is unable to create nicely-sized workgroups. On CPUs, which use threads, this situation does not arise. Further, introducing an if-stmt in the kernel causes "using masked vector remainder loop for the whole iteration space" according to the vectorizer folks. This leads to a severe performance loss in cases where the kernel could be vectorized.
2. The function is_gpu on a handler is true to its name. It allows checking if the kernel is to be executed on a GPU. Its use for turning on/off this optimization is just one of its potential uses.
3. The function is_gpu is not tied to this optimization.